x/bank/keeper/virtual.go

```diff
@@ -0,0 +1,224 @@
+package keeper
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// SendCoinsFromAccountToModuleVirtual sends coins from account to a virtual module account.
+func (k BaseSendKeeper) SendCoinsFromAccountToModuleVirtual(
+	ctx context.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins,
+) error {
+	recipientAcc := k.ak.GetModuleAccount(ctx, recipientModule)
+	if recipientAcc == nil {
+		return errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", recipientModule)
```

Comment:
do we align to panic here?

Reply:
it seems no need to panic if we can return error

Reply:
maybe just align L35

Reply:
done